### PR TITLE
More JSON improvements

### PR DESCRIPTION
**Pull request by adamsitnik**
this gives something around +15-20k RPS for JSON, with the main benefit being the usage of `Utf8JsonWriter`

@sebastienros PTAL

### CODE REVIEW

**Comment by adamsitnik** [comment]
@davidfowl PTAL

File: src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkApplication.Json.cs [hunk 1]
```diff
@@ -7,6 +7,8 @@ namespace PlatformBenchmarks
 {
     public partial class BenchmarkApplication
     {
+        private readonly static uint _jsonPayloadSize = (uint)JsonSerializer.SerializeToUtf8Bytes(new JsonMessage { message = "Hello, World!" }, SerializerOptions).Length;
```
Comment (sebastienros) [inline src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkApplication.Json.cs:10]: @benaadams do you know if that would be an issue? I don't think so.

Comment (benaadams) [reply src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkApplication.Json.cs:10]: No, similar approach in https://github.com/aspnet/Benchmarks/pull/1473 copied from other frameworks; feels a bit of a cheat, but doesn't look to be ruled out

Comment (benaadams) [reply src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkApplication.Json.cs:10]: More specifically actrix is doing this; but also combining all the headers into one static chunk (which  #1473 does)

Comment (benaadams) [reply src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkApplication.Json.cs:10]: Note I copied some of the changes from @adamsitnik's previous PR https://github.com/aspnet/Benchmarks/pull/1466 for #1473

**Comment by adamsitnik** [comment]
@sebastienros @davidfowl can this PR be merged?

**Comment by adamsitnik** [comment]
@sebastienros thank you!

**Pull request merged**